### PR TITLE
Add migration guide for namespaced parameters

### PR DESCRIPTION
**Pull request by lorenabalan**
Signed-off-by: lorenabalan <lorena.balan@quantumblack.com>

## Description

Follow-up to https://github.com/kedro-org/kedro-starters/pull/68 

## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1334"></a>

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -84,6 +84,38 @@
 [Deepyaman Datta](https://github.com/deepyaman), [Lucas Jamar](https://github.com/lucasjamar), [Simon Brugman](https://github.com/sbrugman)
 
 ## Migration guide from Kedro 0.17.* to 0.18.*
+* If you are using any modular pipelines with parameters, make sure they are declared with the correct namespace. See example below:
+
+For a given pipeline:
+```python
+active_pipeline = pipeline(
+    pipe=pipeline_instance,
```
Comment (datajoely) [inline RELEASE.md:92]: I think we need to show what is being overriden here - we don't really have the 'from and too' currently

```suggestion
    pipe=pipeline(
       [
          node(
             func=some_func,
             inputs=["model_input_table", "params:model_options"]
             outputs=...
          )
       ]
    )
```

File: RELEASE.md [hunk 2]
```diff
@@ -84,6 +84,38 @@
...
+    inputs="model_input_table",
+    namespace="candidate_modelling_pipeline",
+)
+```
+
+The parameters should like this:
+
+```diff
```
Comment (datajoely) [inline RELEASE.md:100]: Super cool - I didn't realise you could do this!

**Review by datajoely — CHANGES_REQUESTED** [review]
I think this is on the right track, but I don't think it yer fully covers the previous / future state

File: RELEASE.md (hunk 1 above)
Comment (datajoely) [reply RELEASE.md:92]: This doesn't have to be inline, but I think you need to show what parameters are getting overridden

Comment (lorenabalan) [reply RELEASE.md]: Fair point, done.

**Review by MerelTheisenQB — APPROVED** [review]
LGTM!

**Pull request merged**